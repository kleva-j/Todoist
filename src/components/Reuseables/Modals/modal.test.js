import React from 'react';
import { shallow } from 'enzyme';
import { CustomModal } from '.';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('CustomModal', () => {
  const testProps = {
    width: 500,
    isVisible: true,
    title: 'This is a title',
    handleOk: jest.fn(() => {}),
    handleCancel: jest.fn(() => {}),
  };

  const wrapper = shallow(<CustomModal {...testProps} />);
  const { width, visible, title } = wrapper.props();

  test('should render correctly', () => {
    expect(width).toBe(testProps.width);
    expect(visible).toBe(testProps.isVisible);
    expect(title).toBe(testProps.title);
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
