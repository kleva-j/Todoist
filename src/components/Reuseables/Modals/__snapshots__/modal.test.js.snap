// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomModal should match snapshot 1`] = `ShallowWrapper {}`;
